sdk/python/kfp/components/_python_op.py

```diff
@@ -694,6 +695,118 @@ def add_multiply_two_numbers(a: float, b: float) -> NamedTuple('DummyName', [('s
+        task2 = add_multiply_op(sum_output_ref, 5)
+
+
+    Bigger data should be read from files and written to files.
+    Use the `InputPath` parameter annotation to tell the system that the function wants to consume the corresponding input data as a file. The system will download the data, write it to a local file and then pass the **path** of that file to the function.
+    Use the `OutputPath` parameter annotation to tell the system that the function wants to produce the corresponding output data as a file. The system will prepare and pass the **path** of a file where the function should write the output data. After the function exits, the system will upload the data to the storage system so that it can be passed to downstream components.
+    You can specify the type of the consumed/produced data by specifying the type argument to `InputPath` and `OutputPath`. The type can be a python type or an arbitrary type name string. `OutputPath('CatBoostModel')` means that the function states that the data it has written to a file has type 'CatBoostModel'. `InputPath('CatBoostModel')` means that the function states that it expect the data it reads from a file to have type 'CatBoostModel'. When the pipeline author connects inputs to outputs the system checks whether the types match.
+    Every kind of data can be consumed as a file input. Conversely, bigger data should not be consumed by value as all value inputs pass through the command line.
+
+    Example of a component function declaring file input and output::
+
+        def catboost_train_classifier(
+            training_data_path: InputPath('CSV'),            # Path to input data file of type "CSV"
+            trained_model_path: OutputPath('CatBoostModel'), # Path to output data file of type "CatBoostModel"
+            number_of_trees: int = 100,                      # Small output of type "Integer"
+        ) -> NamedTuple('Outputs', [
+            ('Accuracy', float),  # Small output of type "Float"
+            ('Precision', float), # Small output of type "Float"
+            ('JobUri', 'URI'),    # Small output of type "URI"
+        ]):
+            """Trains CatBoost classification model"""
+            ...
+
+            return (accuracy, precision, recall)
+
+
+    Args:
+        func: The python function to convert
+        base_image: Optional. Specify a custom Docker container image to use in the component. For lightweight components, the image needs to have python 3.5+. Default is the python image corresponding to the current python environment.
+        output_component_file: Optional. Write a component definition to a local file. The produced component file can be loaded back by calling `load_component_from_file` or `load_component_from_uri`.
+        packages_to_install: Optional. List of [versioned] python packages to pip install before executing the user function.
+
+    Returns:
```

Comment:
Great docstring and it makes me wonder, if we should have something similar, e.g., KFP SDK tutorial, in kubeflow/website. The [current one](https://www.kubeflow.org/docs/pipelines/sdk/build-component/) does not include latest functionality/best practices for writing reusable component.

That being said, this should be of low priority. We can do that later.

Reply:
I totally agree. The documentation desperately needs overhaul (as well as the main `README.md`). I've created an issue about switching to reusable components in the documentation: https://github.com/kubeflow/pipelines/issues/2855

This is pretty high priority and we'll try working on this during the doc sprint. We need better docs before the beta launch.